mantidproject/mantid
# Project Recovery: New GUI and new failure GUI

**Description of work.**
I have replaced the original GUI with a more informative GUI in as close to MVP as possible, I also have added a GUI to help people recover as much as possible using the 5 checkpoints so basically they can choose which checkpoint they use.

**To test:**
- Start mantidplot
- Load a workspace
- run ```mantidplot.app.saveRecoveryCheckpoint()```
- run ```Segfault(DryRun=False)```
- Load up mantid again
- Test out the new GUI and see if you can break it (usually a good way to do this is to follow our unscripted testing regime for Project Recovery).

Test out the Recovery Failure GUI:
- Start mantidplot
- Load a workspace
- run ```mantidplot.app.saveRecoveryCheckpoint()```
- run ```Segfault(DryRun=False)```
- Edit the recovery checkpoint's workspace in your .mantid or /%APPADTA%/mantidproject folders, so that it will fail when the script is running.
- Start mantid again and attempt to recover using first GUI
- Then play around with this gui and make sure it does what you think it should etc.



Fixes #23570 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
